include/openthread/thread.h

```diff
@@ -1114,6 +1114,28 @@ void otThreadSetStoreFrameCounterAhead(otInstance *aInstance, uint32_t aStoreFra
  */
 uint32_t otThreadGetStoreFrameCounterAhead(otInstance *aInstance);
 
+/**
+ * Attempts to attach a Wake-up End Device.
```

Comment:
The description `Attempts to attach a Wake-up End Device` easily makes developers think that the WC will attach to WED. When the Thread spec mentions `Attach` it usually says `Attach to a parent`. This function actually sends wake-up frames to the WED and then waiting for the WED to attach to the WC.  Could we rename the API to `otThreadWakeup()`?

Reply:
Hmm. For me, "Attach something [to myself]" and "Attach [myself] to something" mean different things so when I say "Attach WED" it shouldn't be confused with "Attach to WED".

I'm also not sure if `otThreadWakeup` is a future-proof choice as the spec anticipates that in the future the wake-up will be usable to both attach a WED to the Wake-up Coordinator and to a separate Wake-up Parent. I wanted this API to be clear that it means the first use case only and something like `otThreadWakeup` could be added for the more generalized scenario. Any idea how we can add the second use case if we follow your proposal?

Reply:
Yes, we should consider the future usages. The spec says `Wake-up Coordinator (WC) is a role that a Thread Device can play to wake (activate) a Wake -up End Device (WED) and request the WED to connect to a Parent. `. How about changing the API name to `otWcConnect`?

Reply:
@abtink @edmont Any opinions on the naming so that I don't change this multiple times :)

Reply:
Ok. Changing this to `otThreadWakeup`.